File: py/solns/twoSum2SortedArr/twoSum2SortedArr.py

class Solution:
    @staticmethod
    def naive(numbers,target):
        l,r=0,len(numbers)-1
        while l<r:
            tmp = numbers[l]+numbers[r]
            if tmp>target:
                r-=1
            elif tmp<target:
                l+=1
            elif tmp==target:
                break
        return [l+1,r+1]

File: cpp/example/src/TwoSum2SortedArr/twoSum2SortedArr.cpp
#include "twoSum2SortedArr.h"

vector<int> twoSum2SortedArr::naive(
    vector<int>& numbers,int target){
    int l=0,r=numbers.size()-1;
    while (l<r){
        int tmp = numbers[l]+numbers[r];
        if (tmp>target)
            r-=1;
        else if (tmp<target)
            l+=1;
        else if (tmp==target)
            break;
    }
    return vector<int>{l+1,r+1};
}
